ui/components/app/confirm-page-container/confirm-page-container.component.js

```diff
@@ -111,6 +116,14 @@ export default class ConfirmPageContainer extends Component {
     isApprovalOrRejection: PropTypes.bool,
   };
 
+  async componentDidMount() {
+    const { tokenAddress, fromAddress } = this.props;
+    const tokenBalance = await fetchTokenBalance(tokenAddress, fromAddress);
+    this.setState({
+      collectionBalance: tokenBalance?.balance?.words?.[0] || 0,
```

Comment:
Do we know if `tokenBalance = await fetchTokenBalance(tokenAddress, fromAddress); collectionBalance = tokenBalance?.balance?.words?.[0] || 0;` will work for all NFT contracts that have a setApprovalForAll method?

Maybe @adonesky1 knows?

Reply:
@danjm you are correct that this won't work for all NFT contracts. `fetchTokenBalance` uses the `balanceOf` method, which on ERC721's only requires the address to check against as an arg, [but for ERC1155's requires](https://docs.openzeppelin.com/contracts/3.x/api/token/erc1155#IERC1155-balanceOf-address-uint256-) the both the `address` and the `tokenID`. This is because 1155's have the concept of fungible sets of tokenIds so there could be 100 tokens on the contract with `tokenId === 1` and 1000 with `tokenId === 2` all of which are interchangeable. So unfortunately we can't be sure of a users' balance simply on a given ERC1155 contract simply querying directly against the contract (unless we already know all of the tokenIDs they own). We can still achieve this for ERC721 contracts. But for 1155's we'll have to use an indexer like OpenSea to know user balances.

Reply:
My thinking was that we could add in a different ticket/PR the logic here to check if OpenSea API is enabled and use it for getting the balance so we don't block this until we're ready to move the OpenSea API part from behind the NFT feature flag.

Reply:
Sounds right to me!

Reply:
@bschorchit Sure, I will update my PR to show the Total number only for ERC721 standard

Reply:
@bschorchit @adonesky1 
Done, now it will show the total number if the contract follows the ERC721 standard.
You can check it in the storybook also.